src/utils/advance.js

```diff
@@ -0,0 +1,42 @@
+import { Random } from "@woowacourse/mission-utils";
+import { GAME_RULES } from "../constants/index.js";
+
+// 전진 조건 체크
+export const shouldAdvance = (randomNumber) => {
+  return randomNumber >= GAME_RULES.ADVANCE_NUMBER;
+};
+
+// 자동차 전진 로직(전진 카운트 증가)
+export const advanceCars = (carNames, incrementAdvanceCount) => {
+  const { MIN_RANDOM_NUMBER, MAX_RANDOM_NUMBER } = GAME_RULES;
+  carNames.forEach((carName) => {
+    const randomNumber = Random.pickNumberInRange(MIN_RANDOM_NUMBER, MAX_RANDOM_NUMBER);
+
+    if (shouldAdvance(randomNumber)) {
+      incrementAdvanceCount(carName);
+    }
+  });
+};
+
+// 각 자동차의 전진 카운트를 대시로 변환해서 결과 리턴
+export const getAdvanceResult = (carsInfoEntries) => {
+  const result = carsInfoEntries
+    .map(([carName, advanceCount]) => {
+      const dash = GAME_RULES.DASH.repeat(advanceCount);
+      return `${carName} : ${dash}`;
+    })
+    .join("\n");
+
+  return result;
+};
+
+/**
+ * 반복 횟수 만큼 실행 리스트에 있는 함수들을 실행합니다.
+ * @param {number} repeatCount - 반복 횟수
+ * @param {Array<() => void>} executionList - 실행할 함수들의 배열
+ */
+export const executionLoop = (repeatCount, executionList) => {
+  for (let i = 0; i < repeatCount; i++) {
+    executionList.forEach((executionFn) => executionFn());
+  }
+};
```

Comment:
이 부분도 조금 더 순수성을 확보하고 부수 효과를 줄이려면 이런 방식은 어떤가요?

```suggestion
export const executionLoop = (repeatCount, executionList) => {
  Array.from({ length: repeatCount }).map(() =>
    executionList.map((executeFn) => executeFn())
  );
};
```

Reply:
저도 이 부분을 고민했었는데 만약 인자 `repeatCount`가 number 타입이라는 점에서 `for 루프`가 가장 적당했던 것 같아요 

저는 `Array.from`은 배열을 반환하기 때문에 지금 목적과는 조금 다르다는 생각이 드는 것 같아요🤔